### PR TITLE
[IGA] Output Condition

### PR DESCRIPTION
**Pull request by tteschemacher**
A new amazing condition comes to our KRATOS world. Now in a clean branch ;)

**Description**
badabim badabum.

**Changelog**
- Added OutputCondition to IgaApplication

### CODE REVIEW

File: applications/IgaApplication/custom_conditions/output_condition.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        // Integration Points
+        const GeometryType::IntegrationPointsArrayType& integration_points = r_geometry.IntegrationPoints();
+        // Shape function values
+        const Matrix& r_N = r_geometry.ShapeFunctionsValues();
+
+        if (rOutput.size() != integration_points.size())
+            rOutput.resize(integration_points.size());
+        rOutput.clear();
+
+        for (IndexType point_number = 0; point_number < integration_points.size(); point_number++)
```
Comment (dagmawibekel) [inline applications/IgaApplication/custom_conditions/output_condition.cpp]: Better use pre-increment (++point_number)

File: applications/IgaApplication/custom_conditions/output_condition.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        {
+            for (IndexType i = 0; i < nb_nodes; i++)
```
Comment (dagmawibekel) [inline applications/IgaApplication/custom_conditions/output_condition.cpp]: ++i

File: applications/IgaApplication/custom_conditions/output_condition.cpp [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        const SizeType nb_nodes = r_geometry.size();
+
+        // Integration Points
+        const GeometryType::IntegrationPointsArrayType& integration_points = r_geometry.IntegrationPoints();
+        // Shape function values
+        const Matrix& r_N = r_geometry.ShapeFunctionsValues();
+
+        if (rOutput.size() != integration_points.size())
+            rOutput.resize(integration_points.size());
+
```
Comment (dagmawibekel) [inline applications/IgaApplication/custom_conditions/output_condition.cpp:63]: rOutput.Clear() is miising, right?

File: applications/IgaApplication/custom_conditions/output_condition.cpp [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        for (IndexType point_number = 0; point_number < integration_points.size(); point_number++)
```
Comment (dagmawibekel) [inline applications/IgaApplication/custom_conditions/output_condition.cpp]: ++point_number

File: applications/IgaApplication/custom_conditions/output_condition.cpp [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        {
+            rOutput[point_number] = ZeroVector(3);
+            for (IndexType i = 0; i < nb_nodes; i++)
```
Comment (dagmawibekel) [inline applications/IgaApplication/custom_conditions/output_condition.cpp]: ++i

File: applications/IgaApplication/custom_conditions/output_condition.cpp (hunk 3 above)
Comment (tteschemacher) [reply applications/IgaApplication/custom_conditions/output_condition.cpp:63]: Nah, it's just a bit more sofisticated with arrays so I do that in line 66.

File: applications/IgaApplication/custom_conditions/output_condition.cpp (hunk 4 above)
Comment (tteschemacher) [reply applications/IgaApplication/custom_conditions/output_condition.cpp]: Better before, than afterwards ;)

**Comment by tteschemacher** [comment]
Thanks for your review @dagmawibekel

File: applications/IgaApplication/custom_conditions/output_condition.cpp [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+        if (rOutput.size() != integration_points.size())
+            rOutput.resize(integration_points.size());
+
+        for (IndexType point_number = 0; point_number < integration_points.size(); ++point_number)
+        {
+            rOutput[point_number] = ZeroVector(3);
+            for (IndexType i = 0; i < nb_nodes; ++i)
+            {
+                array_1d<double, 3> output_solution_step_value = r_geometry[i].GetSolutionStepValue(rVariable);
```
Comment (philbucher) [inline applications/IgaApplication/custom_conditions/output_condition.cpp]:
```suggestion
                array_1d<double, 3> output_solution_step_value = r_geometry[i].FastGetSolutionStepValue(rVariable);
```

File: applications/IgaApplication/custom_conditions/output_condition.cpp [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+        if (rOutput.size() != integration_points.size())
+            rOutput.resize(integration_points.size());
+        rOutput.clear();
+
+        for (IndexType point_number = 0; point_number < integration_points.size(); ++point_number)
+        {
+            for (IndexType i = 0; i < nb_nodes; ++i)
+            {
+                double output_solution_step_value = r_geometry[i].GetSolutionStepValue(rVariable);
```
Comment (philbucher) [inline applications/IgaApplication/custom_conditions/output_condition.cpp]:
```suggestion
                double output_solution_step_value = r_geometry[i].FastGetSolutionStepValue(rVariable);
```

**Review by philbucher — COMMENTED** [review]
minor but important comment

I also recommend setting up one example for testing the output, since you seem to add more stuff I think it makes sense but ofc up to you :)

File: applications/IgaApplication/custom_conditions/output_condition.cpp (hunk 6 above)
Comment (tteschemacher) [reply applications/IgaApplication/custom_conditions/output_condition.cpp]: Making things fast...

**Comment by tteschemacher** [comment]
> I also recommend setting up one example for testing the output, since you seem to add more stuff I think it makes sense but ofc up to you :)

Thanks for the comment. Within one of the next element tests this condition will definitely appear. As we're having already troubles with the runtime of the tests, I will try to have as few tests as possible...

**Comment by philbucher** [comment]
> Thanks for the comment. Within one of the next element tests this condition will definitely appear. As we're having already troubles with the runtime of the tests, I will try to have as few tests as possible...

@tteschemacher @dagmawibekel I think we should speak abt this, sth seems off there

**Review by philbucher — DISMISSED** [review]
looks good to me :+1:

**Review by dagmawibekel — APPROVED** [review]

**Pull request merged**